### PR TITLE
CAS-434 Fix issue with existing person departure updated domain event

### PR DESCRIPTION
**Pull request by muhammad-elabdulla**
This [PR CAS-434](https://dsdmoj.atlassian.net/browse/CAS-434) is to create a job to fix the current person departure domain events with incorrect event type. 

This job will update the event type in the data field for the current domain events from  **accommodation.cas3.person.departed** to **accommodation.cas3.person.departed.updated**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/DomainEventEntity.kt [hunk 1]
```diff
@@ -108,7 +110,7 @@ data class DomainEventEntity(
   val occurredAt: OffsetDateTime,
   val createdAt: OffsetDateTime,
   @Type(type = "com.vladmihalcea.hibernate.type.json.JsonType")
-  val data: String,
+  var data: String,
```
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/DomainEventEntity.kt:113]: you can use the existing DomainEventRepository.updateData() function, so no need to make this mutable

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/Cas3UpdateDomainEventTypeForPersonDepartureUpdatedJobTest.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            CAS3PersonDepartureUpdatedEvent(
+              id = UUID.randomUUID(),
+              timestamp = Instant.now().randomDateTimeBefore(),
+              eventType = EventType.personDepartureUpdated,
+              eventDetails = CAS3PersonDepartedEventDetailsFactory().produce(),
+            ),
+          ),
+        )
+      }
+    }.take(3).toList()
```
Comment (davidatkinsuk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/Cas3UpdateDomainEventTypeForPersonDepartureUpdatedJobTest.kt:49]: can we please add a domain event of some other type and check it's unmodified after executing the seed job?

**Review by davidatkinsuk — APPROVED** [review]
Looks good, just a few small comments from me to address

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/Cas3UpdateDomainEventTypeForPersonDepartureUpdatedJobTest.kt (hunk 2 above)
Comment (muhammad-elabdulla) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/Cas3UpdateDomainEventTypeForPersonDepartureUpdatedJobTest.kt:52]: Added another domain events

**Pull request merged**